File: hjxarchive/프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.py
def solution(arr1, arr2):
    h = len(arr1)
    w = len(arr2[0])
    r = len(arr2)
    answer = [[0] * w for _ in range(h)]
    for i in range(h):
        for j in range(w):
            for k in range(r):
                answer[i][j] += arr1[i][k]*arr2[k][j]
    return answer

File: limyj108/프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    // (n x m) x (m x k) = (n x k)
    int n = arr1.size(), m = arr2.size(), k = arr2[0].size();
    vector<vector<int>> answer(n, vector<int>(k, 0));
    for (int r = 0; r < n; r++) {
        for (int c = 0; c < k; c++) {
            for (int i = 0; i < m; i++) {
                answer[r][c] += arr1[r][i] * arr2[i][c];
            }
        }
    }
    return answer;
}